### PR TITLE
XS✔ ◾ ♻️ Do you generate dependency graph

### PR DESCRIPTION
**Pull request by babakamyljanovssw**
>
> 1. What triggered this change? (PBI link, Email Subject, conversation + reason, etc)

From Adam
Email: RE: https://www.ssw.com.au/rules/do-you-generate-the-vs-dependency-graph/

> 2. What was changed?

✏️ Updated `do-you-generate-the-vs-dependency-graph` as it was outdated
✏️ Removed `do-you-look-at-the-architecture` as it is duplicate with `do-you-generate-the-vs-dependency-graph` and outdated content
✏️ Changed the rule uri `do-you-generate-the-dependency-graph`

> 3. Did you do pair or mob programming (list names)?

✏️ N/A

### CODE REVIEW

File: rules/do-you-generate-the-dependency-graph/rule.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+* **Additional cost** - After one-month trial, you will need a license to continue using ReSharper
+* **Preferred tool** - Use Rider instead of Visual Studio + ReSharper
+* **Cluttered view** - Complex projects require manual layout adjustments to make the graph readable (no advanced layout tools like Visual Studio has)
+* **Layout** - Only Top to Bottom layout
+
+::: ok  
+![Figure: OK example - The Visual Studio + ReSharper Dependency Graph](sugarlearning-dependency-graph-by-vs-and-resharper.png)  
+:::
+
+When project is complex and large, the graphs generated by these tools can be messy and hard to read. In that case you can try [nDepend](http://www.ndepend.com/) 3rd party solution with the Visual Studio to generate the graph.
```
Comment (jernejk) [inline rules/do-you-generate-the-dependency-graph/rule.md:101]: Let's make this more obvious as optional segment.

```suggestion
## Optional - For very complex (and messy) projects

When project is complex and large, the graphs generated by these tools can be messy and hard to read. In that case you can try [nDepend](http://www.ndepend.com/) 3rd party solution with the Visual Studio to generate the graph.
```

**Review by jernejk — APPROVED** [review]
LGTM. Adam mentioned it should be reviewed by @KostaMadorsky as well.

**Comment by bradystroud** [comment]
## 🤖 Summary from copilot 🤖

This pull request includes updates to various documentation files to standardize references to dependency graphs and remove outdated content. The most important changes include renaming references to the dependency graph, adding a new rule file, and removing obsolete rule files.

Standardizing references to dependency graphs:

* [`rules/architecture-diagram/rule.md`](diffhunk://#diff-1cba977d665767e4e8a654218634b654607b2a1b7b1a162468c1682460a90c91L69-R69): Updated the reference from "VS Dependency Graph" to "Dependency Graph".
* [`rules/do-you-know-what-to-do-about-asp-net-core-aka-asp-net-5-default-dependency-injection/rule.md`](diffhunk://#diff-2df986d190be982e8a690042562819903f084a49d91ad0cfd5b12a7cfb65ec47L18-R18): Updated the related link to "do-you-generate-the-dependency-graph".
* [`rules/the-best-dependency-injection-container/rule.md`](diffhunk://#diff-ab5bdba5e58ddd8d78423f995662ec695ddf17b69632279d649fa48d6cbde9bbL19-R19): Updated related links to "do-you-generate-the-dependency-graph". [[1]](diffhunk://#diff-ab5bdba5e58ddd8d78423f995662ec695ddf17b69632279d649fa48d6cbde9bbL19-R19) [[2]](diffhunk://#diff-ab5bdba5e58ddd8d78423f995662ec695ddf17b69632279d649fa48d6cbde9bbR30) [[3]](diffhunk://#diff-ab5bdba5e58ddd8d78423f995662ec695ddf17b69632279d649fa48d6cbde9bbL42-R44) [[4]](diffhunk://#diff-ab5bdba5e58ddd8d78423f995662ec695ddf17b69632279d649fa48d6cbde9bbR80) [[5]](diffhunk://#diff-ab5bdba5e58ddd8d78423f995662ec695ddf17b69632279d649fa48d6cbde9bbR92) [[6]](diffhunk://#diff-ab5bdba5e58ddd8d78423f995662ec695ddf17b69632279d649fa48d6cbde9bbR104-L107)

Adding new rule file:

* [`rules/do-you-generate-the-dependency-graph/rule.md`](diffhunk://#diff-303b49258cb4217f073a2c921f03d4c86c378ba76240bd1bc2c0d51a30ce5611R1-R110): Added a new rule file detailing the importance of dependency graphs and how to generate them using different tools.

Removing obsolete rule files:

* [`rules/do-you-generate-the-vs-dependency-graph/rule.md`](diffhunk://#diff-f255e9c9a907c32f6a2a506492969c5c01f83e15efe37a1c4c816b0e72fda97dL1-L37): Removed the outdated rule file.
* [`rules/do-you-look-at-the-architecture/rule.md`](diffhunk://#diff-489a97d98036cb7c0b6ac782f51d0405e1aa00405eb402cb98baf241b34c89b3L1-L43): Removed the outdated rule file.

Other minor updates:

* [`categories/software-engineering/rules-to-better-architecture-and-code-review.md`](diffhunk://#diff-7058aef9160b5e47c8a9a31ed70bf46c58faa8f13db34ddef603441b6d5712c3L17-R17): Updated the index to reflect the changes in rule names.
* [`rules/look-at-the-architecture-of-javascript-projects/rule.md`](diffhunk://#diff-e82c84d9b20d58b9d9149e14b78a9b2abad0c54b5ecd2c826b08a1b13ccbd38bL14-R14): Updated related links and improved documentation formatting. [[1]](diffhunk://#diff-e82c84d9b20d58b9d9149e14b78a9b2abad0c54b5ecd2c826b08a1b13ccbd38bL14-R14) [[2]](diffhunk://#diff-e82c84d9b20d58b9d9149e14b78a9b2abad0c54b5ecd2c826b08a1b13ccbd38bL46-R45)
* [`rules/do-you-look-for-grasp-patterns/rule.md`](diffhunk://#diff-108f1e797741cdd14b463c796881cd6bc873c8839dca0cc860c07abecd9e4834L14-R14): Updated related links.

**Review by bradystroud — APPROVED** [review]
Approved because this is a good improvement.

**Review by KostaMadorsky — APPROVED** [review]

**Pull request merged**